### ISSUE TITLE
Switch to WFS 2.0.0

### ISSUE THREAD

**Issue by Roel** [DOV-Vlaanderen/pydov#194]
At some point we might want to switch from current WFS 1.1.0 to WFS 2.0.0.

This would for instance allow to:
* get the maximum feature limit per GetFeature requests (i.e. 10000) from the capabilities of the service instead of hardcoding this in pydov
* work around this limit entirely by using paged requests (which is not supported in WFS 1.1.0)

**Comment by pjhaest** [comment]
👍

**Comment by Roel** [comment]
I think we can/need to implement this in different stages:

* add WFS 2.0.0 GetFeature support via POST in OWSLib (although we could also implement this in pydov as we do now for WFS 1.1.0)
  * @stijnvanhoey started work on this [here](https://github.com/stijnvanhoey/OWSLib/tree/post-getfeature), I suggest focussing our efforts there in preparation of a PR for OWSLib
  * @stijnvanhoey can you lead this?

* add support in OWSLib for FES 2.0 filters, a.o. this means 
  * changing the namespace from http://www.opengis.net/ogc (ogc:) to http://www.opengis.net/fes/2.0 (fes:)
  * changing the ogc:PropertyName to fes:ValueReference
  * @jorisvandenbossche and @johanvdw started looking into this, is there a branch somewhere where we can work on this?
  * who takes the lead on this?

* add support for GML3.2 in the spatial filtering
  * I can look into extending this in pydov, so we can switch once the first two improvements have landed in OWSLib
  * in time we can contribute the spatial filtering with GML3.2 (for WFS 2.0.0) to OWSLib too

Apart from upgrading the stack to WFS 2.0.0, I think it still makes sense to contribute support for WFS 1.1.0 GetFeature via POST as well as the spatial filters for WFS 1.1.0 to OWSLib. I'll create a separate issue for this.

**Comment by stijnvanhoey** [comment]
I'm adding the first PR asap.

**Comment by johanvdw** [comment]
FYI, I have my code to start working on FES filters here:
https://github.com/johanvdw/OWSLib/tree/fes_test

It does not work yet, but if anyone wants to continue working on it, you may have a look.

**Comment by jorisvandenbossche** [comment]
I pushed a branch (and opened https://github.com/DOV-Vlaanderen/pydov/pull/242 from that to more easily see it) with the changes that I made to pydov while testing out WFS 2.0 in pydov. 

Based on that experiment, my impression was that it would not be very hard to already switch to WFS 2.0 in pydov if that would be desired. The main blocker is FES 2.0 support in OWSLib, but for now, that could also be worked around by "fixing" the xml for the filters generated by OWSLib (switching ofc -> fes namespace, replacing PropertyValue with ValueReference). 

For properly improving OWSLib to support FES 2.0, maybe the first step is also to open an issue on the OWSLib side? I didn't directly find one (they have an old one for the spatial filters: https://github.com/geopython/OWSLib/issues/128). Because I suppose there are several ways to tackle this: eg adding a version keyword in the `toXML` as @johanvdw did (or in the main object constructor), or having a separate `fes200.py` file (similarly like they have wfs100.py, wfs110.py, wfs200.py)
